Cargo.toml

```diff
@@ -18,6 +18,7 @@ env_logger = "0.5"
 
 [dependencies]
 plain = "0.2.3"
+memoffset = "0.2.1"
```

Comment:
I haven't reviewed the rest of PR, and I'm definitely interested, but I can tell you that I do not want this (or probably any other) external dependency, and it'll have to go before I consider it.

Reply:
@m4b many thanks for your feedback.
I didn't know that you don't want any more dependency, I've changed the PR to fix that (just a rewrite of `offset_of!` though).